### ISSUE TITLE
False Positive E0107

### ISSUE THREAD

**Issue by Easyoakland** [rust-lang/rust-analyzer#16992]
**rust-analyzer version**: (eg. output of "rust-analyzer: Show RA Version" command, accessible in VSCode via <kbd>Ctrl/⌘</kbd>+<kbd>Shift</kbd>+<kbd>P</kbd>)

0.3.1906-standalone (e4a405f87 2024-03-31)

**rustc version**: (eg. output of `rustc -V`)

rustc 1.76.0 (07dca489a 2024-02-04)

**editor or extension**: (eg. VSCode, Vim, Emacs, etc. For VSCode users, specify your extension version; for users of other editors, provide the distribution if applicable)

VSCode; rust-analyzer v0.3.1906

**relevant settings**: (eg. client settings, or environment variables like `CARGO`, `RUSTC`, `RUSTUP_HOME` or `CARGO_HOME`)

N/A

**repository link (if public, optional)**: (eg. [rust-analyzer](https://github.com/rust-lang/rust-analyzer))

**code snippet to reproduce**: 
```rust
#[allow(unused)]
fn fun<T, I1, I2>(a: I1)
where
    I1: IntoIterator<Item = T>,
    I1::IntoIter: ExactSizeIterator,
    I2: IntoIterator,
{
    let _ = a.into_iter().len().min(0); // expected 0 arguments, found 1 (E0107)
}
```
The above `#[allow(unused)]` is not actually necessary but removes all warnings except the incorrect warning.
The above code builds correctly with `cargo build` and has no error with `cargo check`.
Whatever check reports the error seems to come from "on-the-fly diagnostics" as it occurs/goes away without saving if the problem line or any of the where bounds is commented out .

**Comment by lnicola** [comment]
Duplicate of https://github.com/rust-lang/rust-analyzer/issues/10653.
